From a5fc7af6d0f95c10df20c0c948a1c9f83e7e3b4d Mon Sep 17 00:00:00 2001
From: Devon Loehr <DKLoehr@users.noreply.github.com>
Date: Fri, 1 May 2026 18:19:02 -0400
Subject: [PATCH] c-b: Avoid a build script panic from `.unwrap()` (#1213)

The current code expects CARGO_MANIFEST_DIR to always end in a directory
name, which prevents passing something like ".". We can continue to
support that by unwrapping using `if let` first.

Fixes: https://github.com/rust-lang/compiler-builtins/pull/1187#issuecomment-4360933652
---
 library/compiler-builtins/compiler-builtins/build.rs | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/library/compiler-builtins/compiler-builtins/build.rs b/library/compiler-builtins/compiler-builtins/build.rs
index 748699c4d3e17..e86c784b5c631 100644
--- a/library/compiler-builtins/compiler-builtins/build.rs
+++ b/library/compiler-builtins/compiler-builtins/build.rs
@@ -12,7 +12,10 @@ fn main() {
     // Work around building as part of `builtins-shim`: if only `build.rs` is used, Cargo always
     // considers the build dirty because `builtins-shim/build.rs` does not exist. If only
     // `../c-b/build.rs` is used, the same may happen if not built in the workspace.
-    if cfg.manifest_dir.file_name().unwrap() == "builtins-shim" {
+    // Note: this may be `None` if CARGO_MANIFEST_DIR doesn't end in a directory name (e.g. ".")
+    if let Some(dir) = cfg.manifest_dir.file_name()
+        && dir == "builtins-shim"
+    {
         println!("cargo::rerun-if-changed=../compiler-builtins/build.rs");
     } else {
         println!("cargo::rerun-if-changed=build.rs");
